head/firmware/step_motor.cpp

```diff
@@ -27,4 +27,13 @@ void start_motor_handler(
     handler_class.set_message_queue(queue, completed_queue);
 }
 
-void reset_motor_handler() { handler_class.reset(); }
\ No newline at end of file
+void reset_motor_handler() { handler_class.reset(); }
+
+MoveStatus get_move_status() {
+    auto buffered_move = handler_class.get_buffered_move();
+    MoveStatus msg = {.group_id = buffered_move.group_id,
+                      .seq_id = buffered_move.seq_id,
+                      .remaining_duration = handler_class.get_remaining_ticks(),
```

Comment:
These are _also_ data races, and they _are_ in fact being used. Every so often, you're going to get the motor interrupt firing during the execution of this call, and the data fetched by `get_remaining_ticks` and/or `get_current_position` is going to get corrupted. This access needs to be protected somehow, and protected in a way that cannot significantly delay execution of the interrupt service routine.